classes/XLite/Module/Heartland/Securesubmit/Model/Payment/Securesubmit.php

```diff
@@ -118,43 +127,41 @@ protected function doInitialPayment()
                 if ($cards === array()) {
                     throw new Exception('Stored card cannot be found.');
                 }
-                $token->tokenValue = $cards[0]->getToken();
             } else {
                 $requestMulti = isset($this->request['securesubmit_save_card']) && $this->request['securesubmit_save_card'] === 'save';
             }
 
             if ($this->isCapture()) {
-                $payment = $this->chargeService->charge(
-                    $this->transaction->getValue(),
-                    'usd',
-                    $token,
-                    $cardholder,
-                    $requestMulti,
-                    $details
-                );
+  
+		$payment = $hpstoken->($this->transaction->getValue())
+                ->withCurrency('usd')
+                ->withAddress($address)
+                ->withAllowDuplicates(true)
+				//->withInvoiceNumber($details->invoiceNumber)
+                ->execute();
             } else {
-                $payment = $this->chargeService->authorize(
-                    $this->transaction->getValue(),
-                    'usd',
-                    $token,
-                    $cardholder,
-                    $requestMulti,
-                    $details
-                );
+
+		$payment = $hpstoken->authorize($this->transaction->getValue())
+                ->withCurrency('usd')
+                ->withAddress($address)
+               // ->withInvoiceNumber($details->invoiceNumber)
+                ->withAmountEstimated($this->transaction->getValue())
+                ->withAllowDuplicates(true)
+                ->execute();
+                
             }
 
             if ($requestMulti && $payment->tokenData->responseCode === '0' && $payment->tokenData->tokenValue !== '') {
                 $card = new \XLite\Module\Heartland\Securesubmit\Model\SecuresubmitCreditCard();
                 $this->getEM()->persist($card);
                 $card->setProfileId($this->getProfile()->getProfileId());
-                $card->setToken($payment->tokenData->tokenValue);
+                $card->setToken($hpstoken->token);
```

Comment:
This will break multi-use tokenization as it's storing the single-use token which will be invalid following the initial request. This should be replaced with `$payment->token`. The `if` boolean expression should also be updated to check for this property having a value instead of the previous property.